pymeasure/instruments/toptica/ibeamsmart.py

```diff
@@ -22,99 +22,192 @@
 # THE SOFTWARE.
 #
 
+import logging
+import re
+import time
+
+from pyvisa.errors import VisaIOError
+
 from pymeasure.instruments import Instrument
-from pymeasure.instruments.toptica.adapters import TopticaAdapter
 from pymeasure.instruments.validators import strict_discrete_set, strict_range
 
 
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
 class IBeamSmart(Instrument):
     """ IBeam Smart laser diode
 
-    :param port: pyvisa resource name of the instrument
-    :param baud_rate: communication speed, defaults to 115200
-    :param kwargs: Any valid key-word argument for VISAAdapter
+    :param adapter: pyvisa resource name or adapter instance.
+    :param baud_rate: communication speed, defaults to 115200.
+    :param \\**kwargs: Any valid key-word argument for VISAAdapter.
     """
-    def __init__(self, adapter, baud_rate=115200, **kwargs):
-        if not isinstance(adapter, str):
-            raise TypeError("'adapter' must be a pyvisa resource name string")
+    _reg_value = re.compile(r"\w+\s+=\s+(\w+)")
+
+    def __init__(self, adapter, name="Toptica IBeam Smart laser diode", **kwargs):
+        kwargs.setdefault('read_termination', '\r\n')
+        kwargs.setdefault('write_termination', '\r\n')
         super().__init__(
-            TopticaAdapter(adapter, baud_rate, **kwargs),
-            "toptica IBeam Smart laser diode",
+            adapter,
+            name,
             includeSCPI=False,
+            asrl={'baud_rate': 115200},
```

Comment:
`aslr` dictionary is probably redundant.

Reply:
You mean, I should give the baud rate directly?
As I don't see, where it is redundant (double).

Reply:
This proposed change is affected by this #815. The issue described there made me use the dictionary.